File: Asocioacion/php/cuartoEjercicio/Automovil.php
<?php

//Importe de la Super Clase
require_once('Automotor.php');

class Automovil extends Automotor{

    private string $tipo;
    private string $color;

    public function __construct(string $marca, int $modelo, float $precio, string $tipo, string $color){
        parent::__construct($marca, $modelo, $precio);
        
        $this->tipo = $tipo;
        $this->color = $color;

    }

    //Tipo de automovil
    public function getTipo(){
        return $this->tipo;
    }
    public function setTipo($tipo){
        $this->tipo = $tipo;
    }

    //Color del automovil
    public function getColor(){
        return $this->color;
    }
    public function setColor($color){
        $this->color = $color;
    }

    //Metodo para mostrar datos
    public function mostrarDatos(){
        echo '<h2>'.'<b>'.'AUTOMOVILES'.'</b>'.'</h2>';        
        echo 'La marca del automotor es: '. $this->getMarca() .'<br>';
        echo 'El modelo del automotor es: '.$this->getModelo().'<br>';
        echo 'El precio del automotor es: '.$this->getPrecio().'<br>';
        echo 'El tipo de automovil es: '. $this->tipo.'<br>';
        echo 'El color del automovil es: '. $this->color.'<br>';
    }

}

?>

File: Asocioacion/python/cuartoEjercicio/Automovil.py
#Importe de la clase padre
import Automotor as au

class Automovil(au.Automotor):
    
    c = 0
    
    def __init__(self, marca, modelo ,precio, tipo, color):
        super().__init__(marca, modelo, precio)
        self.tipo = tipo
        self.color = color
        
    #Tipo del Automovil
    def getTipo(self):
        return self.tipo
    
    def setTipo(self, tipo):   
        self.tipo = tipo
        
    #Color del AUtomovil
    def getColor(self):
        return self.color

    def setColor(self, color):
        self.color = color
        
    #Mostrar datos del automovil    
    def mostrarDatos(self):
        print('AUTOMOVILES')
        print('-'*50)
        print(f'La marca del automotor es: {self.marca}')
        print(f'El modelo del automotor es: {self.modelo}')
        print(f'El precio del automotor es: {self.precio}')  
        print(f'El tipo de automovil es: {self.tipo}')
        print(f'El color del automovil es: {self.color}')        